### PR TITLE
Add spec tests for float, decimal and nil values

### PR DESCRIPTION
**Pull request by Anoukh**
## Purpose
Add spec tests for float, decimal and nil values

### CODE REVIEW

File: spec-conformance-tests/05-simple-values/tests/deviations/float.bal [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/test;
+import utils;
+
+// The multiple bit patterns that IEEE 754 treats as NaN are considered to be the same value in Ballerina
+// TODO: Decimal does not support division by zero
+// TODO: == and === not working for float NAN values
+@test:Config {}
+function testFloatingPointNANValuesBroken() {
```
Comment (MaryamZi) [inline spec-conformance-tests/05-simple-values/tests/deviations/float.bal]:
```suggestion
function testFloatingPointNaNValuesBroken() {
```

File: spec-conformance-tests/05-simple-values/tests/deviations/float.bal [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    float f1 = +0.0/0.0;
+    float f2 = -0.0/0.0;
+    //test:assertTrue(f1 == f2, msg = "expected +0.0 and -0.0 to be of same value");
+    test:assertTrue(f1 != f2, msg = "expected +0.0 and -0.0 to be of same value");
```
Comment (MaryamZi) [inline spec-conformance-tests/05-simple-values/tests/deviations/float.bal]: Shall we change the failure messages to reflect the current behaviour?

File: spec-conformance-tests/05-simple-values/tests/float.bal [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    test:assertEquals(d1, d4, msg = "expected decimal values to be equal");
+}
+
+function floatValueDataProvider() returns (float[][]) {
+    return [
+        // DecimalFloatingPointNumber
+        [100e+12, 100E+12, 100e12, 100E12],
+        [100e-2, 100E-2, 1000e-3, 10E-1],
+        [0e2, 0E2, 0e-2, 0e-2],
+        [0.0, .0, 0.0, 0.00], // TODO: Calrify if 0. is valid
```
Comment (MaryamZi) [inline spec-conformance-tests/05-simple-values/tests/float.bal]: We can remove the todos now and maybe add commented out deviation tests for `0.` right?

File: spec-conformance-tests/05-structured-values/tests/lists/arrays.bal [hunk 4]
```diff
@@ -330,9 +331,66 @@ function testArrayInherentTypeViolation() {
...
+    int[] array1 = [];
+    int[] expectedArray = [];
+    test:assertEquals(array1, expectedArray, msg = "expected fixed length array and implied length array to be equal");
+
+    // An array length of !... means that the length of the array is to be implied from the
+    // context; this is allowed in the same contexts where var would be allowed in place of the
+    // type descriptor in which array-length occurs (see “Typed binding patterns”); its meaning
+    // is the same as if the length was specified explicitly.
+    string[3] array2 = ["a", "b", "c"];
+    string[!...] array3 = ["a", "b", "c"];
```
Comment (MaryamZi) [inline spec-conformance-tests/05-structured-values/tests/lists/arrays.bal]: Should we maybe also check if `array3 is string[3]`?

File: spec-conformance-tests/05-structured-values/tests/lists/arrays.bal [hunk 5]
```diff
@@ -330,9 +331,66 @@ function testArrayInherentTypeViolation() {
...
+function testArraySubType() {
+    int[5] fixedArray = [1, 2, 3, 4, 5];
+    any tempArray = fixedArray;
+    test:assertTrue(tempArray is int[], msg = "expected fixed length array to be subtype of growing array");
+
+    utils:BazRecordTwo bRecordTwo = { bazFieldOne: 12.0, bazFieldTwo: "fieldTwo" };
+    any tempRecord = bRecordTwo;
+    test:assertTrue(tempRecord is utils:BazRecord, msg = "expected BazRecordTwo to be subtype of BazRecord");
+
+
```
Comment (MaryamZi) [inline spec-conformance-tests/05-structured-values/tests/lists/arrays.bal]: extra new line?
